policy/reviewer/reviewer.go

```diff
@@ -189,9 +189,14 @@ func FindRandomRequesters(ctx context.Context, prctx pull.Context, result common
 		}
 
 		if len(possibleReviewers) > 0 {
-			logger.Debug().Msgf("Found %d total candidates for review after removing author and non-collaborators; randomly selecting %d", len(possibleReviewers), child.ReviewRequestRule.RequiredCount)
-			randomSelection := selectRandomUsers(child.ReviewRequestRule.RequiredCount, possibleReviewers, r)
-			usersToRequest = append(usersToRequest, randomSelection...)
+			if child.ReviewRequestRule.AddEveryone {
```

Comment:
I can't comment on the line but `FindRandomRequesters` is no longer an accurate name for this function.

If we do go with the `mode` option approach, a nice refactor might be to wrap these in strategy functions so that this becomes a `switch` on the mode that calls the correct strategy for the mode.

Reply:
We you thinking of a function to wrap `SelectReviewers`?

Reply:
I was thinking of something like this:

```golang
switch mode {
case common.RequestModeRandomUsers:
  usersToRequest = selectRandomly(users)
case common.RequestModeAllUsers:
  usersToRequest = selectAll(users)
}
```

The benefit here is minimal since the selection logic is simple. It's more valuable as complexity increases and you might be returning teams _and_ users or doing other logic.

Reply:
We will probably need to change up some of the selection if we want to support team requests, since the gh api expects the team names to be passed as a separate parameter: https://developer.github.com/v3/pulls/review_requests/#input

Maybe we can note this for now as work that should definitely be done later, but only once we've got a good usecase for it?